docs/styleguide.md

```diff
@@ -1,78 +1,179 @@
+
+type Props = {
+  pressed: boolean
 }
 
-```
+const blueIfPressed = ({ pressed }: Props) => css`
+  background-color: ${pressed ? cVar('colorBackgroundPrimary') : 'initial'};
+`
 
-## Styling components
+const Container = styled.div<Props>`
+  display: flex;
+  ${blueIfPressed};
+`
 
-We mainly use `@emotion/styled` to style components, if there is some style pattern or "snippet" that is repeated between multiple components, we like to use the `css` prop from `@emotion/react` to do so.
-When possible, we like to use variables from our theme, which is imported from `@/shared/theme`, theme components should be distructured before being used. When possible, *always* use values from the theme as it helps keep the app consistent.
+const Button = styled.button<Props>`
+  padding: ${sizes(4)};
+  ${blueIfPressed};
+`
+```
 
-Here is a kitchen sink example:
+### Conditional styles
 
-```javascript
-// YourComponent.styles.tsx
-import styled from "@emotion/styled"
-import { css } from "@emotion/react"
+In a lot of cases, styles need to be based on some condition, for example different colors based on component variant, etc. In those cases we use props provided by the `emotion` library. One important aspect to keep an eye on is to avoid keeping conditional styles inside raw string - this will make our tooling not properly understand the style, disabling benefits of linters, IDE features, etc. Example:
 
-import { colors, sizes } from "@/shared/theme"
+```tsx
+import { css } from '@emotion/react'
+import styled from '@emotion/styled'
 
+type Props = {
+  enabled: boolean
+}
 
-const blueIfPressed = (props) => css`
- Background-color: ${props => props.pressed ? colors.blue[500] : ''};
+// don't do this:
+const BadExample = styled.div<Props>`
+  // in this example opacity is passed in a raw string and will not be properly interpreted by tooling
+  ${({ enabled }) => (!enabled ? 'opacity: 0.5;' : '')};
+`
 
+// do this:
+const disabledCss = css`
+  opacity: 0.5;
 `
-export const Container = styled.div`
-	display: flex;
-	background-color: ${colors.gray[500]};
+
+const GoodExample = styled.div<Props>`
+  // in this example conditional style is defined in a separate css block and will be properly parsed by tooling
+  ${({ enabled }) => !enabled && disabledCss};
 `
+```
+
+## Components
+
+### Directory structure
+
+Every reusable component should be placed inside `src/components`. If any of the component categories (directories starting with underscore, e.g. `_buttons`) is a good fit, the component should be placed there. Every component should have its own directory named the same as the component. The directory should contain the following files:
+
+- `Component/`
+  - `index.ts` - re-exporting anything needed
+  - `Component.tsx` - component code
+  - _(optional)_ `Component.styles.ts` - any styles needed for the component
+  - _(optional)_ `Component.stories.tsx` - Storybook stories exploring different use cases/variants of the component
```

Comment:
Maybe we should add an extra line here for a unit test?
eg:
- _(optional)_ `Component.test.tsx` - Component's unit tests

Reply:
Let's do that once we actually introduce unit tests